File: app/schemas/inventory.py
from pydantic import BaseModel
from typing import Optional, List
from decimal import Decimal
from datetime import datetime

# Input para crear un ajuste manual
class AdjustmentCreate(BaseModel):
    variant_id: str
    quantity: Decimal # Positivo para entrada, Negativo para salida
    reason: str       # "Compra", "Merma", "Inventario Inicial"
    notes: Optional[str] = None
    branch_id: Optional[int] = None

class TransferCreate(BaseModel):
    variant_id: str
    from_branch_id: int
    to_branch_id: int
    quantity: Decimal
    reason: str
    notes: Optional[str] = None

# Output para leer el Kardex
class MovementRead(BaseModel):
    id: int
    movement_type: str
    qty_change: Decimal
    qty_after: Decimal
    reference: Optional[str]
    created_at: datetime
    user_name: str # Extraemos el nombre del usuario

    class Config:
        from_attributes = True

File: frontend/src/api/inventory.ts
import client from './client'

export interface KardexMovement {
  id: number
  movement_type: string
  qty_change: number
  qty_after: number
  reference: string | null
  created_at: string
  user_name: string
}

export interface AdjustmentCreate {
  variant_id: string
  quantity: number
  branch_id: number
  reason: string
  notes?: string
}

export interface TransferCreate {
  variant_id: string
  from_branch_id: number
  to_branch_id: number
  quantity: number
  reason?: string
}

export const inventoryApi = {
  getKardex: async (variantId: string, branchId?: number): Promise<KardexMovement[]> => {
    const { data } = await client.get<KardexMovement[]>(`/inventory/kardex/${variantId}`, {
      params: branchId ? { branch_id: branchId } : undefined,
    })
    return data
  },

  createAdjustment: async (payload: AdjustmentCreate): Promise<KardexMovement> => {
    const { data } = await client.post<KardexMovement>('/inventory/adjust', payload)
    return data
  },

  transferStock: async (payload: TransferCreate): Promise<{ message: string }> => {
    const { data } = await client.post<{ message: string }>('/inventory/transfer', payload)
    return data
  },
}
